velox/vector/VectorStream.h

```diff
@@ -82,6 +88,34 @@ class VectorSerializer {
   virtual void flush(OutputStream* stream) = 0;
 };
 
+/// Serializer that writes a subset of rows from a single RowVector to the
+/// OutputStream.
+///
+/// Each call to serialize will serialize the specified range(s) of `vector` and
+/// write them to the output stream.
+class BatchVectorSerializer {
```

Comment:
nit: would it make sense to add a comment to each Serializer interface explaining the usecase where each would make sense (eg: small set of rows vs a large batch of rows that can potentially benefit from this)